routing/datasource.go

```diff
@@ -528,6 +531,69 @@ type routeTable struct {
 	created       time.Time
 }
 
+func fadeInState(now time.Time, duration time.Duration, detected time.Time) (time.Duration, bool) {
+	rel := now.Sub(detected)
+	return rel, rel > 0 && rel < duration
+}
+
+func fadeIn(now time.Time, duration time.Duration, exponent float64, detected time.Time) float64 {
+	rel, fadingIn := fadeInState(now, duration, detected)
+	if !fadingIn {
+		return 1
+	}
+
+	return math.Pow(float64(rel)/float64(duration), exponent)
+}
+
+func goodEndpoint(rnd *rand.Rand, ep *LBEndpoint, route *Route) bool {
+	now := time.Now()
+	f := fadeIn(
+		now,
+		route.LBFadeInDuration,
+		route.LBFadeInExponent,
+		ep.Detected,
+	)
+	goodFadeIn := f == 1 || rnd.Float64() < f
+
+	return goodFadeIn
+}
+
+func GoodEndpointerGoroutine(routes []*Route, quit <-chan struct{}, tick <-chan time.Time) {
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano())) // #nosec
+
+	for {
+		select {
+		case <-quit:
+			return
+		case <-tick:
+			for _, ri := range routes {
+				if ri.Route.BackendType != eskip.LBBackend {
+					continue
+				}
+
+				if len(ri.Route.LBEndpoints) == 0 {
+					log.Errorf("failed to post-process LB route: %s, no endpoints defined", ri.Id)
+					continue
+				}
+
+				newGoodEndpoints := []LBEndpoint{}
```

Comment:
Makes sense to allocate enough memory right away
`make([]LBEndpoint, len(ri.LBEndpoints))`

Reply:
or ` make([]LBEndpoint, 0, len(ri.LBEndpoints))`  in case you use `append()` (read cap vs len)